cpp/tests/strings/array_tests.cpp

```diff
@@ -0,0 +1,234 @@
+        thrust::make_transform_iterator( h_strings.begin(), [] (auto str) { return str!=nullptr; }));
+    cudf::size_type start = 3;
+    cudf::size_type end = GetParam();
+    std::vector<const char*> h_expected;
+    if( end > start )
+    {
+        for( cudf::size_type idx=start; (idx < end) && (idx < (cudf::size_type)h_strings.size()); ++idx )
+            h_expected.push_back( h_strings[idx] );
+    }
+    auto strings_view = cudf::strings_column_view(strings);
+    auto results = cudf::strings::detail::slice(strings_view,start,end);
+
+    cudf::test::strings_column_wrapper expected( h_expected.begin(), h_expected.end() );
+         //thrust::make_transform_iterator( h_expected.begin(), [] (auto str) { return str!=nullptr; }));
+    cudf::test::expect_columns_equal(*results,expected);
+}
+
+TEST_P(SliceParmsTest, SliceAllNulls)
+{
+    std::vector<const char*> h_strings{ nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr };
+    cudf::test::strings_column_wrapper strings( h_strings.begin(), h_strings.end(),
+        thrust::make_transform_iterator( h_strings.begin(), [] (auto str) { return str!=nullptr; }));
+    cudf::size_type start = 3;
+    cudf::size_type end = GetParam();
+    std::vector<const char*> h_expected;
+    if( end > start )
+    {
+        for( cudf::size_type idx=start; (idx < end) && (idx < (cudf::size_type)h_strings.size()); ++idx )
+            h_expected.push_back( h_strings[idx] );
+    }
+    auto strings_view = cudf::strings_column_view(strings);
+    auto results = cudf::strings::detail::slice(strings_view,start,end);
+    cudf::test::strings_column_wrapper expected( h_expected.begin(), h_expected.end(),
+         thrust::make_transform_iterator( h_expected.begin(), [] (auto str) { return str!=nullptr; }));
+    cudf::test::expect_columns_equal(*results,expected);
+}
+
+TEST_P(SliceParmsTest, SliceAllEmpty)
+{
+    std::vector<const char*> h_strings{ "", "", "", "", "", "", "" };
+    cudf::test::strings_column_wrapper strings( h_strings.begin(), h_strings.end(),
+        thrust::make_transform_iterator( h_strings.begin(), [] (auto str) { return str!=nullptr; }));
+    cudf::size_type start = 3;
+    cudf::size_type end = GetParam();
+    std::vector<const char*> h_expected;
+    if( end > start )
+    {
+        for( cudf::size_type idx=start; (idx < end) && (idx < (cudf::size_type)h_strings.size()); ++idx )
+            h_expected.push_back( h_strings[idx] );
+    }
+    auto strings_view = cudf::strings_column_view(strings);
+    auto results = cudf::strings::detail::slice(strings_view,start,end);
+    cudf::test::strings_column_wrapper expected( h_expected.begin(), h_expected.end() );
+         //thrust::make_transform_iterator( h_expected.begin(), [] (auto str) { return str!=nullptr; }));
+    cudf::test::expect_columns_equal(*results,expected);
+}
+
+INSTANTIATE_TEST_CASE_P(SliceParms, SliceParmsTest,
+                        testing::ValuesIn(std::array<cudf::size_type,3>{5,6,7}));
```

Comment:
Minor suggestion:
same code is repeated except `h_strings`. Can `h_strings` be made as another parameter?

Reply:
I'm not sure why git marked this as a new file. I think I only changed the extension from '.cpp' to '.cu' but may have also removed a header include or two.